src/nodes/NodeConnectionManager.ts

```diff
@@ -508,6 +546,9 @@ class NodeConnectionManager {
     await this.quicServer.stop({ force: true });
     await this.quicSocket.stop({ force: true });
     await this.rpcServer.stop({ force: true });
+    if (this.mdns[running]) {
+      await this.mdns.stop();
+    }
```

Comment:
Since stop is idempotent I think you can just call stop without checking if its running. 

Usually we only check if something is running prior to stopping if its happening in the background. But in this case the `mdns` could just not be running. So I'm not sure, maybe @CMCDragonkai can weigh in.

Reply:
This is no longer relevant due to MDNS being an injected optional dep